src/itr_evaluator.ml

```diff
@@ -1220,7 +1233,15 @@ and evaluate_expr (context : 'a context) (e : expr) : record_item =
     | Some context ->
       (match String_map.get v context.local_vars with
       | Some (Record_item ri) -> evaluate_record_item ri
-      | _ -> Rec_value e))
+      | Some (Msg { msg; _ }) ->
+        evaluate_record_item (context.get_field msg [])
+      | None -> 
+        (match context.implicit_message with 
+          | Some {msg; _} when String.equal v "msg_data" ->
```

Comment:
This seems to be needed where the variable `msg_data` doesn't exist in context, but should be treated as the root of the incoming message (e.g. in some function applications - particularly in hofs it seems).

There may be a better way of solving this. The 'correct' solution would probably be adjusting references to `msg_data` to use `MessageValue` instead of `ObjectProperty` during the AST transpilation. It would appear that this comes in [here](https://github.com/imandra-ai/ipl-worker/blob/7e39eedb27218895d337a04cd78a4b27b2299893/imandra-test-runner/itr_transpile.ml#L570-L590), but I can't see where `"msg_data"` is referenced - so this might actually need to be fixed higher up?

Reply:
Discussed offline and have reverted. The variable should be set [here](https://github.com/imandra-ai/ipl-worker/blob/7e39eedb27218895d337a04cd78a4b27b2299893/imandra-test-runner/itr_transpile.ml#L528-L534), it just wasn't happening for the special multi-constraint triage scripts. Fixed separately in https://github.com/imandra-ai/ipl-worker/commit/5dfec10cb59e38d6d8103c7f8080101fecfbd547